### DISCUSSION TITLE
IDialogService.ShowDialog resulting in ex "Value does not fall within the expected range."

### DISCUSSION THREAD

**Discussion by sigmarsson** [PrismLibrary/Prism#2492] [Uno Platform]
1. Registering my dialog, which is a `UserControl` :

`containerRegistry.RegisterDialog<SettingsDialog>(RegionNames.Settings);` 

2. And trying to show later but an excpetion is caught :

![image](https://user-images.githubusercontent.com/43340959/122795151-51ebec80-d2bd-11eb-9232-527b1660a502.png)
  
Please see the ex details and advise how to resolve those errors :

- [x] 1. The first one is claming the paramter is incorrect. Can I pass in a `null` ? 
- [x] 2. Second : `"This element is already associated with a XamlRoot, it cannot be associated with a different one until it is removed from the previous XamlRoot."` Am I expected to set the dialog's `XamlRoot` ? I encountered earlier such WinUI related obstacles but I wonder if Prism's dialog service handles this or not.

### Exception Details

![image](https://user-images.githubusercontent.com/43340959/122795582-c1fa7280-d2bd-11eb-8bc3-ec4c8794f95c.png)

**Comment by brianlagunas** [comment]
You need to set that. See this issue https://github.com/PrismLibrary/Prism/issues/2472
